source/material_model/visco_plastic.cc

```diff
@@ -758,12 +838,29 @@ namespace aspect
                                    "See, for example, Thieulot, C. (2011), PEPI 188, pp. 47-68. "
                                    "\n\n"
                                    "The user has the option to linearly reduce the cohesion and "
-                                   "internal friction angle as a function of the finite strain invariant. "
-                                   "The finite strain invariant may be calculated through particles or the compositional "
-                                   "field finite strain invariant plugin (see \\cite{buiter2008dissipation} benchmark). "
-                                   "In either case, the user specifies a compositional field for the finite "
-                                   "strain invariant, which must be the first listed compositional field "
-                                   "in the parameter file."
+                                   "internal friction angle as a function of the finite strain magnitude. "
+                                   "The finite strain invariant or full strain tensor is calculated through "
+                                   "compositional fields within the material model. This implementation is "
+                                   "identical to the compositional field finite strain plugin and cookbook "
+                                   "described in the manual (author: Gassmoeller). If the user selects to track "
+                                   "the finite strain invariant ($e_ii$), a single compositional field tracks "
+                                   "the value derived from $e_ii^t = (e_ii)^(t-1) + edot_ii*dt, where $t$ and $t-1$ "
+                                   "are the current and prior time steps, $edot_ii$ is the second invariant of the "
+                                   "strain rate tensor and $dt$ is the time step size. In the case of the "
+                                   "full strain tensor $F$, the finite strain magnitude is derived from the "
+                                   "second invariant of the symmetric stretching tensor $L$, where "
+                                   "$L = F * [F]^T$. In either case, the user specifies a single compositional "
```

Comment:
"In either case" sounds confusing to me, because you don't do the same thing in either case. Either you specify a single field or multiple fields